code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeUtils.java

```diff
@@ -175,4 +179,102 @@ static String getRequestEventId(final Event event) {
         }
         return requestEventId;
     }
+
+    /**
+     * Dispatches an event to track propositions with type {@value
+     * OptimizeConstants.EventType#OPTIMIZE} and source {@value
+     * OptimizeConstants.EventSource#REQUEST_CONTENT}.
+     *
+     * <p>No event is dispatched if the provided {@code xdm} is null or empty.
+     *
+     * @param xdm {@code Map<String, Object>} containing the XDM data for the proposition
+     *     interactions.
+     */
+    static void trackWithData(final Map<String, Object> xdm) {
+        if (OptimizeUtils.isNullOrEmpty(xdm)) {
+            Log.debug(
+                    OptimizeConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Failed to dispatch track propositions request event, input xdm is null or"
+                            + " empty.");
+            return;
+        }
+
+        final Map<String, Object> eventData = new HashMap<>();
+        eventData.put(
+                OptimizeConstants.EventDataKeys.REQUEST_TYPE,
+                OptimizeConstants.EventDataValues.REQUEST_TYPE_TRACK);
+        eventData.put(OptimizeConstants.EventDataKeys.PROPOSITION_INTERACTIONS, xdm);
+
+        final Event edgeEvent =
+                new Event.Builder(
+                                OptimizeConstants.EventNames.TRACK_PROPOSITIONS_REQUEST,
+                                OptimizeConstants.EventType.OPTIMIZE,
+                                OptimizeConstants.EventSource.REQUEST_CONTENT)
+                        .setEventData(eventData)
+                        .build();
+
+        MobileCore.dispatchEvent(edgeEvent);
+    }
+
+    /**
+     * Generates a map containing XDM formatted data for `Experience Event - OptimizeProposition
+     * Interactions` field group from this `OptimizeProposition` offer and given
+     * `experienceEventType`.
+     *
+     * <p>
+     *
+     * <p>The method returns null if the proposition reference within the offer is released and no
+     * longer valid.
+     *
+     * @param experienceEventType [String] containing the event type for the Experience Event
+     * @return `Map<String, Object>` containing the XDM data for the proposition interaction.
+     */
+    protected static Map<String, Object> generateInteractionXdm(
+            final String experienceEventType,
+            final List<OptimizeProposition> pendingDisplayedPropositions,
+            final Boolean shouldAppendOfferIds) {
```

Comment:
In case of ODE, we can get a list of offers inside a single proposition. This list might be shown inside a recycler view and only few of them might be visible on the screen at a time. 
The app developer may choose to call `Offer.displayed()` only when the user has scrolled to that section in the recycler view, which would give a accurate metric for number of offer displayed. 
he way we have it here adds all items from the proposition to the display event list, thereby incrementing display metric for all offers, which is not correct